lib/src/domain/communication/api_exception.dart

```diff
@@ -1,4 +1,6 @@
-class ApiException implements Exception {}
+import 'package:allthenews/src/domain/communication/all_the_news_exception.dart';
+
+abstract class ApiException extends AllTheNewsException {}
```

Comment:
błędy z firebase to też api exception tak logicznie rzecz biorąc :)

Reply:
to `NyTimesApiException` i `FirebaseApiException` ?

Reply:
Może zostawić ApiException jako podstawę, bo jednak te błędy, które używamy pod NyTimes są raczej typowymi błędami w komunikacji po http i w przypadku większości api taka struktura jaką mamy będzie pasować do większości api (a możliwe, że jakieś inne podepniemy). Sam Firebase ma specyficzne odpowiedzi, bo jednak bazują na domenie i wynikają z kluczy w postaci stringów, co jest raczej specyficzne tylko dla firebase dlatego wydaje mi się, że fajnie jakby zostało  ApiException jako trzon. A niech `AuthenticationApiException` je może po prostu może rozszerza? 🤔